apps/api/src/app/endpoints/public/public.controller.ts

```diff
@@ -61,43 +65,155 @@ export class PublicController {
+        );
+      }
+
+      // Add holding filters (symbol + dataSource)
+      // Each holding needs both DATA_SOURCE and SYMBOL filters
+      if (accessFilter.holdings && accessFilter.holdings.length > 0) {
+        accessFilter.holdings.forEach((holding) => {
+          portfolioFilters.push(
+            {
+              id: holding.dataSource,
+              type: 'DATA_SOURCE' as const
+            },
+            {
+              id: holding.symbol,
+              type: 'SYMBOL' as const
+            }
+          );
+        });
+      }
+    }
+
     const [
       { createdAt, holdings, markets },
       { performance: performance1d },
       { performance: performanceMax },
       { performance: performanceYtd }
     ] = await Promise.all([
       this.portfolioService.getDetails({
+        filters: portfolioFilters.length > 0 ? portfolioFilters : undefined,
         impersonationId: access.userId,
         userId: user.id,
         withMarkets: true
       }),
       ...['1d', 'max', 'ytd'].map((dateRange) => {
         return this.portfolioService.getPerformance({
           dateRange,
+          filters: portfolioFilters.length > 0 ? portfolioFilters : undefined,
           impersonationId: undefined,
           userId: user.id
         });
       })
     ]);
 
+    // Filter out only the base currency cash holdings
+    const baseCurrency =
+      user.settings?.settings.baseCurrency ?? DEFAULT_CURRENCY;
+    const filteredHoldings = Object.fromEntries(
+      Object.entries(holdings).filter(([, holding]) => {
+        // Remove only cash holdings that match the base currency
+        const isCash = holding.assetSubClass === AssetSubClass.CASH;
+        const isBaseCurrency = holding.symbol === baseCurrency;
+        return !(isCash && isBaseCurrency);
+      })
+    );
+
+    // Use filters for activities, but exclude DATA_SOURCE/SYMBOL filters
+    // if there are multiple holdings (the service can't handle multiple symbol filters)
+    const hasMultipleHoldingFilters =
+      accessFilter?.holdings && accessFilter.holdings.length > 1;
+
+    const activityFilters = portfolioFilters.filter((filter) => {
+      // Always include ACCOUNT, ASSET_CLASS, TAG filters
+      if (
+        filter.type === 'ACCOUNT' ||
+        filter.type === 'ASSET_CLASS' ||
+        filter.type === 'TAG'
+      ) {
+        return true;
+      }
+
+      // Include DATA_SOURCE and SYMBOL only if there's a single holding filter
+      if (
+        !hasMultipleHoldingFilters &&
+        (filter.type === 'DATA_SOURCE' || filter.type === 'SYMBOL')
+      ) {
+        return true;
+      }
+
+      return false;
+    });
+
     const { activities } = await this.orderService.getOrders({
+      filters: activityFilters.length > 0 ? activityFilters : undefined,
       includeDrafts: false,
       sortColumn: 'date',
       sortDirection: 'desc',
-      take: 10,
+      take: hasMultipleHoldingFilters ? 1000 : 10, // Get more if we need to filter manually
```

Comment:
❓

Reply:
I did not understood how filtering was passed to orderService. So I did a quick test to try different things. This is a leftover. I removed it and fixed another issue that I had below.

> ❓